eth2-ssv/beacon/prysmgrpc/propose.go

```diff
@@ -0,0 +1,96 @@
+package prysmgrpc
+
+import (
+	"context"
+	"github.com/bloxapp/ssv/slotqueue"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// GetProposalData implements Beacon interface
+func (b *prysmGRPC) GetProposalData(ctx context.Context, slot uint64, duty slotqueue.Duty) (*ethpb.BeaconBlock, error) {
+	randaoReveal, err := b.signRandaoReveal(ctx, slot, duty)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to sign randao reveal")
+	}
+
+	block, err := b.validatorClient.GetBlock(ctx, &ethpb.BlockRequest{
+		Slot:         slot,
+		RandaoReveal: randaoReveal,
+		Graffiti:     b.graffiti,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get block")
+	}
+
+	return block, nil
+}
+
+// SignProposal implements Beacon interface
+func (b *prysmGRPC) SignProposal(ctx context.Context, block *ethpb.BeaconBlock, duty slotqueue.Duty) (*ethpb.SignedBeaconBlock, error) {
+	// TODO: Check this
+	/*if err := b.preBlockSignValidations(ctx, block); err != nil {
+		return nil, errors.Wrapf(err, "failed block safety check for slot %d", block.Slot)
+	}*/
+
+	sig, err := b.signBlock(ctx, block, duty)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not sign block")
+	}
+
+	// TODO: Check this
+	/*if err := b.postBlockSignUpdate(ctx, block, domain); err != nil {
+		return nil, errors.Wrapf(err, "failed post block signing validations for slot %d", blk.Block.Slot)
+	}*/
```

Comment:
There is no need for a post block sign update, as that was an old approach we used in Prysm. Instead, we can just sign, then check slashable, then submit